### PR TITLE
Cjc/clamp

### PR DESCRIPTION
**Pull request by ccjincong**
## Motivation and Context
fix clamp、clamp_min、clamp_max、clamp_scalar、clamp_max_scalar、clamp_min_scalar

## Description
fix clamp

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend/device_configs.py [hunk 1]
```diff
@@ -1471,7 +1387,7 @@
             args=[
                 {
                     "ins": ['input'],
-                    "dtype": [Skip(Dtype.float32),Skip(Dtype.float64),Skip(Dtype.float16),Skip(Dtype.int16),Skip(Dtype.int32),Skip(Dtype.int64),Skip(Dtype.uint8),Skip(Dtype.int8),Skip(Dtype.bool)],
+                    "dtype": [Skip(Dtype.float16)],
```
Comment (hellozmz) [inline impl/ascend/device_configs.py:1390]: 在 impl/ascend/convert_config.yaml 中对数据类型转换下，把float16也一起支持了吧

File: impl/ascend/functions/clamp.cpp [hunk 2]
```diff
@@ -16,71 +17,121 @@ diopiError_t diopiClamp(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopi
...
+    diopiTensorHandle_t minTmp, maxTmp, boolOut;
+    makeTensorLike(ctx, &minTmp, input, dtype);
+    makeTensorLike(ctx, &maxTmp, input, dtype);
+    makeTensorLike(ctx, &boolOut, input, diopi_dtype_bool);
+
+    std::vector<int64_t> sizes;
+    diopiSize_t diopiShape;
+    diopiGetTensorShape(input, &diopiShape);
+    std::vector<int64_t> shapeTmp(diopiShape.data, diopiShape.data + diopiShape.len);
+    sizes = std::move(shapeTmp);
```
Comment (hellozmz) [inline impl/ascend/functions/clamp.cpp:34]: 直接使用ascendTensor，可以直接获取到shape。

File: impl/ascend/functions/clamp.cpp [hunk 3]
```diff
@@ -16,71 +17,121 @@ diopiError_t diopiClamp(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopi
...
+
     if (max != nullptr) {
-        runner.addInput(max);
-    } else {
-        diopiTensorHandle_t maxTmp;
-        makeTensorLike(ctx, &maxTmp, input, dtype);
-        if (isIntegralType(dtype)) {
-            fillTensor(ctx, maxTmp, static_cast<float>(INT_MAX));
+        if (!sizes.empty()) {
+            AclOpRunner<2, 1>("BroadcastTo", ctx).addInput(max, dtype).addConstInput(sizes).addOutput(maxTmp).run();
```
Comment (hellozmz) [inline impl/ascend/functions/clamp.cpp:48]: 已经有broadcast函数了，用broadcast函数替换下

File: impl/ascend/functions/clamp.cpp [hunk 4]
```diff
@@ -16,71 +17,121 @@ diopiError_t diopiClamp(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopi
...
+        fillTensor(ctx, maxTmp, std::numeric_limits<double>::max());
     }
-    runner.addOutput(out).run();
-    return diopiSuccess;
-}
 
-diopiError_t diopiClampInp(diopiContextHandle_t ctx, diopiTensorHandle_t input, diopiConstTensorHandle_t min, diopiConstTensorHandle_t max) {
-    return diopiClamp(ctx, input, input, min, max);
+    diopiLt(ctx, boolOut, maxTmp, minTmp);
+    diopiMaskedFill(ctx, minTmp, minTmp, boolOut, maxTmp);
```
Comment (hellozmz) [inline impl/ascend/functions/clamp.cpp:57]: 加下注释信息描述下，为什么要将数据进行调整。

File: impl/ascend/device_configs.py (hunk 1 above)
Comment (ccjincong) [reply impl/ascend/device_configs.py:1354]: 底层无法支持

**Review by hellozmz — APPROVED** [review]

**Review by yangbofun — APPROVED** [review]

**Pull request merged**